src/ynEIGEN/RedemptionAssetsVault.sol

```diff
@@ -0,0 +1,227 @@
+    //--------------------------------------------------------------------------------------
+
+    function deposit(uint256 amount, address asset) external {
+        if (!assetRegistry.assetIsSupported(IERC20(asset))) revert AssetNotSupported();
+
+        balances[asset] += amount;
+        IERC20(asset).safeTransferFrom(msg.sender, address(this), amount);
+
+        emit AssetDeposited(asset, msg.sender, amount);
+    }
+
+    /** 
+     * @notice Calculates the current redemption rate of ynETH to ETH.
+     * @return The current redemption rate as a uint256.
+     */
+    function redemptionRate() public view returns (uint256) {
+        return ynEigen.previewRedeem(YNETH_UNIT);
+    }
+
+    /** 
+     * @notice Returns the total amount of assets available for redemption.
+     * @return _availableRedemptionAssets The available unit-of-account-denominated balance as a uint256.
+     */
+    function availableRedemptionAssets() public view returns (uint256 _availableRedemptionAssets) {
+
+        IERC20[] memory assets = assetRegistry.getAssets();
+
+        uint256 len = assets.length;
+        for (uint256 i = 0; i < len; ++i) {
+            IERC20 asset = assets[i];
+            uint256 balance = balances[address(asset)];
+            if (balance > 0) _availableRedemptionAssets += assetRegistry.convertToUnitOfAccount(asset, balance);
+        }
+    }
+
+    /** 
+     * @notice Transfers a specified amount of redemption assets to a given address.
+     * @param to The recipient address of the assets.
+     * @param amount The amount of assets to transfer.
+     * @dev Requires the caller to be the redeemer and the contract to not be paused.
+     */
+    function transferRedemptionAssets(address to, uint256 amount, bytes calldata /* data */) public onlyRedeemer whenNotPaused nonReentrant {
+        uint256 balance = availableRedemptionAssets();
+        if (balance < amount) revert InsufficientAssetBalance(ETH_ASSET, amount, balance);
+
+        IERC20[] memory assets = assetRegistry.getAssets();
+        for (uint256 i = 0; i < assets.length; ++i) {
+            IERC20 asset = assets[i];
+            uint256 assetBalance = balances[address(asset)];
+            if (assetBalance > 0) {
+                uint256 assetBalanceInUnit = assetRegistry.convertToUnitOfAccount(asset, assetBalance);
+                if (assetBalanceInUnit >= amount) {
+                    uint256 reqAmountInAsset = assetRegistry.convertFromUnitOfAccount(asset, amount);
+                    IERC20(asset).safeTransfer(to, reqAmountInAsset);
+                    balances[address(asset)] -= reqAmountInAsset;
+                    break;
+                } else {
+                    IERC20(asset).safeTransfer(to, assetBalance);
+                    balances[address(asset)] = 0;
+                    amount -= assetBalanceInUnit;
+                }
+            }
+        }
+
+        emit AssetTransferred(ETH_ASSET, msg.sender, to, amount);
```

Comment:
the even should have the first value of amount, this one is potentially 0

Reply:
https://github.com/yieldnest/yieldnest-protocol/pull/164/commits/9895e89b6b9cf7b04283f666b7fe4773883ee84b